### PR TITLE
adding docker_worker subnet_ids and logic to allow static vpc

### PR DESCRIPTION
**Pull request by Gregory-Pereira**
cc @lmilbaum not 100% this will work but this is my best first shot at this

### CODE REVIEW

**Comment by lmilbaum** [comment]
CI is broken

**Comment by Gregory-Pereira** [comment]
Changes:

- Switching to pinned os runner version while here
- adding python version to reduce flakes in precommit
- original feature: allow users to pass static VPC for both agent and works and subnet ID for just workers

**Review by lmilbaum — CHANGES_REQUESTED** [review]
Have you tested it with the two options: once with vpc_id configured and one without?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,11 +18,11 @@ jobs:
     strategy:
       matrix:
         terraform: [ 1.6.1 ]
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
[the remark this answers is not in the document]
Comment (Gregory-Pereira) [reply .github/workflows/ci.yml:21]: Ack i can remove it work now. No I have not tested it yet in any way, will do that now.

**Comment by Gregory-Pereira** [comment]
Proven this works!

**Review by lmilbaum — APPROVED** [review]

**Pull request merged**